api/handlers.go

```diff
@@ -86,25 +95,30 @@ func GetByIDSessionHandlerFunc(cache Cache, getVarsFunc GetVarsFunc) http.Handle
 }
 
 // GetByEmailSessionHandlerFunc returns a function that retrieves a session by ID from the cache
-func GetByEmailSessionHandlerFunc(cache Cache, getVarsFunc GetVarsFunc) http.HandlerFunc {
+func GetByEmailSessionHandlerFunc(sessionCache Cache, getVarsFunc GetVarsFunc) http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
 		ctx := r.Context()
 		email := getVarsFunc(r)["Email"]
 
-		s, err := cache.GetByEmail(email)
+		s, err := sessionCache.GetByEmail(email)
 		if err != nil {
-			writeErrorResponse(ctx, w, err.Error(), err, http.StatusNotFound)
+			if err == cache.ErrSessionNotFound {
+				writeErrorResponse(ctx, w, sessionNotFoundErr, err, http.StatusNotFound)
```

Comment:
looking at the `writeErrorResponse` function for this call - when err != nil it will log details of the error message, but not the `sessionNotFoundErr`. I wonder if the log should also contain the msg string?